### ISSUE TITLE
InputText with big frame padding weird behavior

### ISSUE THREAD

**Issue by mooko-dev** [ocornut/imgui#3781]
Hello!
I want to draw the InputText frame with big horizontal inner frame padding.
Code:
```
        ImGui::Begin("Window");

        ImGui::PushStyleVar(ImGuiStyleVar_FramePadding, ImVec2(50, ImGui::GetStyle().FramePadding.y));
        static char str[256];
        ImGui::InputText("Text", str, sizeof(str));
        ImGui::PopStyleVar();

        ImGui::End();
```

Weird behavior is:
1) Padding's ignored if the text width greater than the frame width.
![example](https://user-images.githubusercontent.com/78313188/106611564-08923f00-6560-11eb-9c93-6f112ab45f77.gif)
2) If the text doesn't fit in the frame and I continue typing, it doesn't immediately move the cursor to the end of the text, unlike when there is no padding
Without padding:
![without_padding](https://user-images.githubusercontent.com/78313188/106612416-febd0b80-6560-11eb-82cb-87fecebe6ac4.gif)
With padding:
![with_padding](https://user-images.githubusercontent.com/78313188/106612457-054b8300-6561-11eb-9453-c7153674a7f6.gif)
3) Frame padding applies only to the left side. Is it possible to apply frame padding for the right side too?

**Comment by ocornut** [comment]
Padding is not designed to be large.

> (1) Padding's ignored if the text width greater than the frame width.

We need to overlap that padding section to signify that there is content on the of the box, otherwise there would be no way to get that information at a glance.

> (2) If the text doesn't fit in the frame and I continue typing, it doesn't immediately move the cursor to the end of the text, unlike when there is no padding

That looks like a bug and I'll look into fixing it now.

> (3) Frame padding applies only to the left side. Is it possible to apply frame padding for the right side too?

Same reason as (1).

**Comment by ocornut** [comment]
To workaround (1) and (3) you might be able display a larger frame yourself, then while calling `InputText()` set the FrameBg color to zero with `PushStyleColor()`. It might or not work for you depending on why you need such unusual amount of padding.

**Comment by mooko-dev** [comment]
> To workaround (1) and (3) you might be able display a larger frame yourself, then while calling `InputText()` set the FrameBg color to zero with `PushStyleColor()`. It might or not work for you depending on why you need such unusual amount of padding.

It's good idea, thanks!

**Comment by ocornut** [comment]
Pushed d1a9efdd8f90b09df1ae380403992e8a997f7ede with 3 fixes:
- Input cursor tracking was slightly off (point 2)
- Padding/clipping was slightly different for multiline inputs, it now does the same as single-line, as explained above.
- Mouse cursor tracking was slightly off by FramePadding.y vertically on multiline inputs (not mentioned here)

**Comment by ocornut** [comment]
This added a small regression #4794 which was fixed by 072caa4
